services/base.js

```diff
@@ -58,43 +58,54 @@ module.exports = class BaseService {
     return [];
   }
 
-  static register(camp, handleRequest) {
-    const serviceClass = this; // In a static context, "this" is the class.
-
+  static get _regex() {
     // Regular expressions treat "/" specially, so we need to escape them
-    const escapedPath = serviceClass.uri.format.replace(/\//g, '\\/');
+    const escapedPath = this.uri.format.replace(/\//g, '\\/');
     const fullRegex = '^' + escapedPath + '.(svg|png|gif|jpg|json)$';
+    return new RegExp(fullRegex);
+  }
+
+  static _namedParamsForMatch(match) {
+    // Assume the last match is the format, and drop match[0], which is the
+    // entire match.
+    const captures = match.slice(1, -1);
+
+    if (this.uri.capture.length !== captures.length) {
+      throw new Error(
```

Comment:
I do not know all the plans for services rewrite but in my opinion it would be good idea to add test for this fragment and for other untested code in this base class.

Reply:
Agreed this is a great idea.